include/parser/compile-state.hpp

```diff
@@ -34,6 +34,7 @@ class MacroDirective;
  \brief Contains all relevant information for a compile run.
  */
 struct CompileState {
+  using TranslateablePtr = Translateable::TranslateablePtr;
```

Comment:
This feature has been in review before your parser fix was merged, so some of your mentions should already be fixed (most of string-parser.hpp and all of integer-parser.hpp).
I will correct the rest as well as the suggestions in #102.